heareval/tasks/dcase2016_task2.py

```diff
@@ -0,0 +1,144 @@
+    },
+    # TODO: FIXME
+    # Want different for train and test?
+    "sample_duration": 120.0,
+    "partitions": [
+        {"name": "train", "max_files": 10},
+        {"name": "test", "max_files": 10},
+    ],
+}
+
+
+class ConfigureProcessMetaData(luigi_util.WorkTask):
+    """
+    Custom metadata pre-processing. Creates a metadata csv
+    file that will be used by downstream luigi tasks to curate the final dataset.
+    TODO: It would be nice to have a better description of what this pattern is
+    """
+
+    outfile = luigi.Parameter()
+    train = luigi.TaskParameter()
+    test = luigi.TaskParameter()
+
+    def requires(self):
+        return {
+            "train": self.train,
+            "test": self.test,
+        }
+
+    @staticmethod
+    def get_rel_path(root: Path, item: pd.DataFrame) -> Path:
+        # Creates the relative path to an audio file given the note_str
+        audio_path = root.joinpath("audio")
+        filename = f"{item}.wav"
+        return audio_path.joinpath(filename)
+
+    @staticmethod
+    def slugify_file_name(filename: str) -> str:
+        return f"{slugify(filename)}.wav"
+
+    def get_split_metadata(self, split: str, split_path_str: str) -> pd.DataFrame:
+        logger.info(f"Preparing metadata for {split}")
+
+        split_path = (
+            Path(self.requires()[split].workdir)
+            .joinpath(split)
+            .joinpath(split_path_str)
+        )
+
+        metadatas = []
+        for annotation_file in split_path.glob("annotation/*.txt"):
+            metadata = pd.read_csv(
+                annotation_file, sep="\t", header=None, names=["start", "end", "label"]
+            )
+            sound_file = (
+                str(annotation_file)
+                .replace("annotation", "sound")
+                .replace(".txt", ".wav")
+            )
+            assert os.path.exists(sound_file)
+            metadata = metadata.assign(relpath=sound_file)
+
+            metadata = metadata.assign(
+                slug=lambda df: df.relpath.apply(self.slugify_file_name)
+            )
+            metadata = metadata.assign(partition=lambda df: split)
+            metadata = metadata.assign(
+                filename_hash=lambda df: df["slug"].apply(
+                    luigi_util.filename_to_int_hash
+                )
+            )
+            metadatas.append(metadata)
+
+        return pd.concat(metadatas)[luigi_util.PROCESSMETADATACOLS]
+
+    def run(self):
+        # Get metadata for each of the data splits
+        process_metadata = pd.concat(
+            # DCASE 2016 uses funny pathing, so we just hardcode the desired
+            # paths
+            # Note that from our training data, we only use DCASE 2016 dev data.
+            # Their training data is short monophonic events.
+            [
+                self.get_split_metadata(
+                    "train", "dcase2016_task2_train_dev/dcase2016_task2_dev/"
```

Comment:
should we use os.path.join here to support windows as well?

Reply:
Whaddya mean?